toolchain/check/import_cpp.cpp

```diff
@@ -172,19 +194,36 @@ class CarbonClangDiagnosticConsumer : public clang::DiagnosticConsumer {
           CARBON_DIAGNOSTIC(CppInteropParseWarning, Warning, "{0}",
                             std::string);
           CARBON_DIAGNOSTIC(CppInteropParseError, Error, "{0}", std::string);
-          context_->emitter().Emit(
+          auto builder = context_->emitter().Build(
               SemIR::LocId(info.import_ir_inst_id),
               info.level == clang::DiagnosticsEngine::Warning
                   ? CppInteropParseWarning
                   : CppInteropParseError,
               info.message);
+          for (;
+               i + 1 < diagnostic_infos_.size() &&
+               diagnostic_infos_[i + 1].level == clang::DiagnosticsEngine::Note;
+               ++i) {
+            const ClangDiagnosticInfo& note_info = diagnostic_infos_[i + 1];
+            CARBON_DIAGNOSTIC(CppInteropParseNote, Note, "{0}", std::string);
+            builder.Note(SemIR::LocId(note_info.import_ir_inst_id),
+                         CppInteropParseNote, note_info.message);
+          }
+          builder.Emit();
           break;
         }
       }
     }
+    diagnostic_infos_.clear();
   }
 
  private:
+  // Callback to flush Clang diagnostics into the Carbon diagnostics engine.
+  struct DiagnosticFlusher {
+    CarbonClangDiagnosticConsumer* consumer_;
```

Comment:
Consider explicitly initializing to nullptr to avoid accidentally having it unset.
```suggestion
    CarbonClangDiagnosticConsumer* consumer_ = nullptr;
```
Alternatively, we can make it a class with a private data member and initialize the data member in the constructor.

Reply:
This field is non-nullable, so initializing it to null seems very misleading. Given the tiny scope of this helper struct, I don't really think it's worth providing additional encapsulation here. We only have one instance of this and that seems unlikely to change, so the risk of missing the initialization seems pretty low.